src/models/referral_store.rs

```diff
@@ -4,9 +4,15 @@ use primitive_types::{H160, H256};
 use std::collections::HashMap;
 use std::sync::Mutex;
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum Referral {
+    Address(Option<H160>),
+    TryToFetchXTimes(u64),
```

Comment:
Could you elaborate on the meaning of this here? I am guessing its a configuration value like some number telling you how many times to attempt something. If that is the case, then this seems like an awkward place to keep it...